File: analyse/test_algos.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class EraserTest {
    @Test
    public void testSolution() {
        assertEquals("666,thenumberofthebeast", Erase.erase("666 , the number of the beast"));
        assertEquals("06  07651970", Erase.erase("06  07 65 19 70 "));
        assertEquals("OK", Erase.erase("OK"));
        assertEquals("Coucou  JM  B", Erase.erase(" Cou cou  J M  B"));
        assertEquals("", Erase.erase(""));
        assertEquals("", Erase.erase(" "));
        assertEquals("  ", Erase.erase("  "));
        assertEquals("   ", Erase.erase("   "));
        assertEquals("    ", Erase.erase("    "));
        assertEquals("   f", Erase.erase("   f "));
        assertEquals("a", Erase.erase(" a"));
        assertEquals("       ", Erase.erase("       "));
    }
}

File: analyse/test_algos.py
def erase(cc):
    pass


def test ():
    if erase ('06   07 65 19 70 ') == '06   07651970' :
        print ('OK test 1')
    if erase ('666, the number of the beast') == '666,thenumberofthebeast' :
        print ('OK test 2')
    if erase (' Cou cou  J M  B') == 'Coucou  JM  B' :
        print ('OK test 3')
    if erase ('') == '' :
        print ('OK test 4')
    if erase (' ') == '' :
        print ('OK test 5')
    if erase ('  ') == '  ' :
        print ('OK test 6')
    if erase ('   ') == '   ' :
        print ('OK test 7')
    if erase ('    ') == '    ' :
        print ('OK test 8')
    if erase ('   f ') == '   f' :
        print ('OK test 9')
    if erase (' a') == 'a' :
        print ('OK test 10')
    if erase ('       ') == '       ' :
        print ('OK test 11')

test ()